### PR TITLE
Add `check` and `ci-check` targets to Make

### LINKED ISSUE #896: Add `make check` and `make ci-check`
We have two test suites, one in C++, another in Rust. Both are build by `make test`, but to run them, you need to invoke different commands (`./test/test` and `cargo test`). There are two possible ways to run them, which I'd call…

1. **local**: `cargo test && ./test/test`. This is what we advise in the README, and that's what I personally run locally. This incantation fails as soon as some tests start failing — very useful when you're able to immediately tweak the code and re-run the tests.

2. **CI**: run `cargo test`, remember its exit code, run `./test/test`, remember its exit code, exit with the highest of the two codes. This obtains as much data as possible, making the most out of the job's run.

    This should also pass `--no-fail-fast` flag to Cargo, otherwise it stops running tests as soon as some start failing (https://github.com/rust-lang/cargo/issues/1904); that partially defeats the point for us.

I think both scenarios should be Make targets, which we can advertise in the README.

### PR DESCRIPTION
**Pull request by Minoru**
Fixes #896.

Reviews are welcome! If no issues are raised, I'll merge this in three days.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -136,11 +136,14 @@ check out [our style guide](doc/internal/code-style.markdown).
...
-	$ make -j5 PROFILE=1 all test  # 5 is CPU cores + 1, to parallelize the build
-	$ (cd test && TMPDIR=/dev/shm ./test --order rand) && cargo test
-
-Note the use of ramdisk as `TMPDIR`: some tests create temporary files, which
-slows them down if `TMPDIR` is on HDD or even SSD.
+	$ TMPDIR=/dev/shm make -j5 PROFILE=1 check
+
+The "5" here is the number of CPU cores in your machine *plus one*. This
+parallelises the build. Rust tests already utilize as many cores as they can,
+but if you want to limit them, use `RUST_TEST_THREADS` environment variable.
```
Comment (der-lyse) [inline README.md:143]: I reckon there should to be an article: …use **the** `RUST_TEST_THREADS` environment variable.

File: README.md [hunk 2]
```diff
@@ -136,11 +136,14 @@ check out [our style guide](doc/internal/code-style.markdown).
...
+`/dev/shm` is a "ramdisk", i.e. a virtual disk stored in the RAM. The tests
+create a lot of temporary files, and benefit from fast storage; a ramdisk is
+better than even as SSD.
```
Comment (der-lyse) [inline README.md:146]: Sounds a bit clumsy to me: …is even better than an SSD.

**Review by der-lyse — CHANGES_REQUESTED** [review]
Couldn't try it out yet, but looks alright. I just got two very minor things.

**Comment by der-lyse** [comment]
Both new targets work here.

File: README.md (hunk 1 above)
Comment (Minoru) [reply README.md]: Thanks! Fixed in 9685313d961ffa756165f8fce37a572c0fd85f56.

File: README.md (hunk 2 above)
Comment (Minoru) [reply README.md]: Ah, I even had a typo there (should be s/as/an/). But your version is indeed better, so I used it in 9685313d961ffa756165f8fce37a572c0fd85f56. Thanks!

**Review by der-lyse — APPROVED** [review]
Perfect!

**Comment by Minoru** [comment]
Okay, thanks! I'm squashing those fixups then.

**Pull request merged**